build/sphinx/mongoc_common.py

```diff
@@ -1,7 +1,7 @@
 import os
```

Comment:
```suggestion
from __future__ import annotations

import os
```

The `__future__.annotations` import is back-compatible to Python 3.7 and causes deferred-evaluation of type annotations, see: https://peps.python.org/pep-0563/. I try to include this in almost every module, but it often slips by me. With this addition, the `str | None` is parsed but not evaluated. See also: https://docs.python.org/3/library/__future__.html#id1

Reply:
Neat. Thank you for the explaination.